src/Tribe/Aggregator/Processes/Service_Provider.php

```diff
@@ -16,6 +16,8 @@ public function register() {
 		tribe_register( 'events-aggregator.record-items', 'Tribe__Events__Aggregator__Record__Items' );
 		tribe_register( 'events-aggregator.processes.import-events', 'Tribe__Events__Aggregator__Processes__Import_Events' );
 		tribe_singleton( 'events-aggregator.queue-control', 'Tribe__Events__Aggregator__Processes__Queue_Control' );
+		$batch_imports = new Tribe__Events__Aggregator__Processes__Batch_Imports();
+		tribe_singleton( Tribe__Events__Aggregator__Processes__Batch_Imports::class, $batch_imports );
```

Comment:
Not a good idea to trigger the initialization of the class here, it defeats the purpose of using a service provider.
```suggestion
		tribe_singleton( Tribe__Events__Aggregator__Processes__Batch_Imports::class, Tribe__Events__Aggregator__Processes__Batch_Imports::class );
```

Reply:
Great question, however this is actually better as the instance is provided to the service provider instead of a string to be resolved, more details can be found here: https://github.com/moderntribe/the-events-calendar/pull/3306#discussion_r523962403